api/v1/ocsinitialization_types.go

```diff
@@ -25,16 +25,6 @@ import (
 
 // OCSInitializationSpec defines the desired state of OCSInitialization
 type OCSInitializationSpec struct {
-	// EnableCephTools toggles on whether or not the ceph tools pod
-	// should be deployed.
-	// Defaults to false
-	// +optional
-	EnableCephTools bool `json:"enableCephTools,omitempty"`
-
-	// Tolerations if specified set toolbox ceph tools pod tolerations
-	// Defaults to empty
-	// +optional
-	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
```

Comment:
These are breaking changes to an existing CR interface and we should not do them this way!

If the desire is for breaking changes to an existing CR in release `n` then you will have to do the following:

In release `n`:
1.  introduce a new CR version (v2) with the breaking changes 
2. support backward compatibility for this release
3. Mark the old CR version as deprecated  

In release `n + 1`
1. remove support for the V1 CR 

All of this needs to be done because a CR interface is a public interface that our internal and external customers' automation might rely on. One example is the ODF Managed service deployer.

Reply:
@agarwal-mudit FYI

Reply:
For now, I have reverted these changes to ocsinitialization, The EnableCephTools & Tolerations will stay here in ocsinitialization for now. 
So basically for now we have now two duplicate ways of enabling
    cephToolBox,
    1. Via Ocsinitialization
    2. Via Storagecluster
    
    If done by patching Ocsinitialization then the toolbox pod will
    still, stay there even after the storage cluster is gone.
    
    If done via patching Storagecluster then whenever the storagecluster
    goes away the toolbox pod goes away automatically as the storagecluster
    is added as an owner to that.

We will figure out an upgrade/deprecation path from OCSInitialization to StorageCluster control later.